server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -550,7 +668,7 @@ private void constructStarTree(TreeNode node, int startDocId, int endDocId) thro
 
         // Construct all non-star children nodes
         node.childDimensionId = childDimensionId;
-        Map<Long, TreeNode> children = constructNonStarNodes(startDocId, endDocId, childDimensionId);
+        Map<Long, InMemoryTreeNode> children = constructNonStarNodes(startDocId, endDocId, childDimensionId);
```

Comment:
Major comment :
The key is dimension value.
For documents with '-1' value,
Here in this method, "-1" key will have the associated node.

And in next 'constructStarNode' method , the key '-1' will be overridden with star node again.